src/jit/lower.h

```diff
@@ -178,15 +178,17 @@ class Lowering : public Phase
     {
         assert(GenTree::OperIsBinary(tree->OperGet()));
 
-        GenTree* op1 = tree->gtGetOp1();
-        GenTree* op2 = tree->gtGetOp2();
+        GenTree* const op1 = tree->gtGetOp1();
+        GenTree* const op2 = tree->gtGetOp2();
 
-        if (tree->OperIsCommutative() && tree->TypeGet() == op1->TypeGet())
+        const bool op1Legal = tree->OperIsCommutative() && (tree->TypeGet() == op1->TypeGet());
+        const bool op2Legal = tree->TypeGet() == op2->TypeGet();
```

Comment:
Changing the legality tests to use the size rather than the type results in some minor improvements, all of which are the inverse of the diff I reported above. For example:

```diff
        cmp      r8d, 9
        jne      SHORT G_M65468_IG04
-       mov      rax, gword ptr [rsp+28H]
-       or       r9, rax
+       or       r9, qword ptr [rsp+28H]
        jne      SHORT G_M65468_IG04
        mov      eax, 264
```